server.go

```diff
@@ -235,27 +244,35 @@ func (sc *serverConn) writeReset(strm uint32, code ErrorCode) {
 	sc.writer <- fr
 }
 
-func (sc *serverConn) writeGoAway(strm uint32, code ErrorCode) {
+func (sc *serverConn) writeGoAway(strm uint32, code ErrorCode, message string) {
 	ga := AcquireFrame(FrameGoAway).(*GoAway)
 
 	fr := AcquireFrameHeader()
-	fr.SetStream(strm)
```

Comment:
Why removing the SetStream? Maybe you don't want to close the connection (for example, in the timeout case). And if strm = 0, the stream will be zero anyways. I prefer to keep the `fr.SetStream`. What are your thoughts?

Reply:
In the specification:
```
The GOAWAY frame applies to the connection, not a specific stream. An endpoint MUST treat a GOAWAY frame with a stream identifier other than 0x0 as a connection error (Section 5.4.1) of type PROTOCOL_ERROR.
```

So for me, we need to keep this to 0x0.

Reply:
Ok. It makes sense. That's fine.